sources/api/host-containers/src/main.rs

```diff
@@ -446,10 +459,15 @@ async fn run() -> Result<()> {
     let mut failed = 0usize;
     let host_containers = get_host_containers(args.socket_path).await?;
     for (name, image_details) in host_containers.iter() {
-        // Continue to handle other host containers if we fail one
-        if let Err(e) = handle_host_container(name, image_details) {
-            failed += 1;
-            error!("Failed to handle host container '{}': {}", &name, e);
+        // handle all host containers during startup
+        // handle the host container that has settings changed during restart
+        if changed_settings.len() == 0
```

Comment:
This check (BTW you can use is_empty()) could be moved into the `verify_change_for_container` function (which I suggest renaming to `is_restart_required`)... so that function would start like this:

```rust
fn is_restart_required(changed_settings: &String, container_name: &str) -> bool {
if changed_settings.empty() {
  // if change_settings is empty, it means that Bottlerocket is booting.
  // all containers need to be started
  return true;
}
// ...

```

It would eliminate another conditional that takes some time to understand when reading.